### PR TITLE
[PORT/FEATURE] Animated closet doors

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Ports animated closet doors from Yogstation/Skyrat
https://github.com/yogstation13/Yogstation/pull/9706
https://github.com/Skyrat-SS13/Skyrat-tg/pull/6767

Closets still open and close instantly, but do play an animation when their door is opened.

## Why It's Good For The Game
Animated closet doors look good.

## Images of changes

https://github.com/user-attachments/assets/161fd4dc-a730-42d8-b1aa-675fe516d3ca

## Testing
Open and close a bunch of lockers
More testing will be needed

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl: monster860
add: Animated closet doors
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -25,13 +26,28 @@
 	var/material_drop = /obj/item/stack/sheet/metal
 	var/material_drop_amount = 2
 	var/transparent
+	var/secure = FALSE
+
+		/// The overlay for the closet's door
```
Comment (lewcc) [inline code/game/objects/structures/crates_lockers/closets.dm:31]:
```suggestion
	/// The overlay for the closet's door
```

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 2]
```diff
@@ -25,13 +26,28 @@
...
+	var/obj/effect/overlay/closet_door/door_obj
```
Comment (lewcc) [inline code/game/objects/structures/crates_lockers/closets.dm:32]: Make sure this gets cleaned up on Destroy()

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 3]
```diff
@@ -25,13 +26,28 @@
...
+	/// The maximum angle the door will be drawn at
+	var/door_anim_angle = 136
+	/// X position of the closet door hinge
+	var/door_hinge_x = -6.5
+	/// Amount of time it takes for the door animation to play
+	var/door_anim_time = 2.0 // set to 0 to make the door not animate at all
+
+	var/enable_door_overlay = TRUE
+	var/has_opened_overlay = TRUE
+	var/has_closed_overlay = TRUE
```
Comment (lewcc) [inline code/game/objects/structures/crates_lockers/closets.dm:46]: Document these, please

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 4]
```diff
@@ -42,6 +58,83 @@
...
+		door_obj = new
+	var/default_door_icon = "[closed_door_sprite || icon_state]_closed"
+	vis_contents += door_obj
+	door_obj.icon = icon
+	door_obj.icon_state = default_door_icon
+	is_animating_door = TRUE
+	var/num_steps = door_anim_time / world.tick_lag
+
+	for(var/step in 0 to num_steps)
+		var/angle = door_anim_angle * (closing ? 1 - (step/num_steps) : (step/num_steps))
```
Comment (lewcc) [inline code/game/objects/structures/crates_lockers/closets.dm:97]:
```suggestion
		var/angle = door_anim_angle * (closing ? 1 - (step / num_steps) : (step / num_steps))
```

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 5]
```diff
@@ -96,21 +188,27 @@
...
-	update_icon()
+		return
+	if(opened)
+		return
+	welded = FALSE
+	locked = FALSE
 	playsound(loc, open_sound, open_sound_volume, TRUE, -3)
+	opened = TRUE
+	// if(!dense_when_open)
+	// 	set_density(FALSE)
```
Comment (lewcc) [inline code/game/objects/structures/crates_lockers/closets.dm:201]: Was this meant to be put somewhere else?
```suggestion
```

**Review by S34NW — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Comment by Monkeke2** [comment]
Hey, unsure if this has been reported, but unlocking a personal closed bugs it out. 
Screenshot 2024-10-14 192622
Screenshot 2024-10-14 192640
No idea where I should report this so here I go.

**Pull request merged**